php/class-post-type.php

```diff
@@ -549,4 +548,37 @@ public function filter_user_has_cap( $allcaps, $caps ) {
 
 		return $allcaps;
 	}
+
+	/**
+	 * Publish snapshot changes when snapshot post is being published.
+	 *
+	 * @param int      $post_id Post id.
+	 * @param \WP_Post $post Post object.
+	 */
+	public function publish_snapshot( $post_id, $post ) {
+		if ( ! is_a( $this->snapshot_manager->customize_manager, '\WP_Customize_Manager' ) ) {
+			require_once( ABSPATH . WPINC . '/class-wp-customize-manager.php' );
+			$this->snapshot_manager->customize_manager = new \WP_Customize_Manager();
+		}
+		$snapshot_content = json_decode( $post->post_content, true );
```

Comment:
Instead of accessing the `post_content` directly, it would be better to do:

```php
$snapshot_content = $this->snapshot_manage->post_type->get_post_content( $post );
```